zerocopy-derive/tests/ui/padding_check.stderr

```diff
@@ -0,0 +1,21 @@
+error[E0277]: the trait bound `HasPadding<AsBytesStruct, true>: ShouldBe<false>` is not satisfied
+  --> tests/ui/padding_check.rs:15:10
+   |
+15 | #[derive(AsBytes)]
+   |          ^^^^^^^ the trait `ShouldBe<false>` is not implemented for `HasPadding<AsBytesStruct, true>`
+   |
+   = help: the trait `ShouldBe<VALUE>` is implemented for `HasPadding<T, VALUE>`
+   = help: see issue #48214
+   = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
```

Comment:
Uh oh, it looks like this feature actually just breaks the `where` bound approach entirely: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=2ad3a2c4527a3f8b5b3baa33a4c0dfa6

Reply:
If you actually use it, then you get an error: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=0adc598549ab840bce52c61705142053

This is true of all generic bounds, so if the plan is to eventually make this compatible with generics then I think it's the right approach. Maybe for non-generic types we could emit a constant to force immediate resolution? Something like:

```rs
fn must_be_as_bytes<T: AsBytes>() {}
const _: fn() = must_be_as_bytes::<Foo>;
```

That would force resolution for concrete types.

Reply:
OK yeah that sounds good to me! I didn't realize that this would fail at resolution time. Forcing immediate resolution sounds like the right approach.

Reply:
Added a constant usage for concrete types.